tests/deps/setup_rejson.sh

```diff
@@ -0,0 +1,60 @@
+#!/bin/bash
+
+# Function to run a command, and only if it fails, print stdout and stderr and then exit
+run_command() {
+  output=$(eval "$@" 2>&1)
+  status=$?
+  if [ $status -ne 0 ]; then
+    echo "$output"
+    exit $status
+  fi
+}
+
+# Set the default variables
+CURR_DIR=`pwd`
+ROOT=${ROOT:=$CURR_DIR}  # unless ROOT is set, assume it is the current directory
+BINROOT=${BINROOT:=${ROOT}/bin/linux-x64-release}
+
+JSON_BRANCH=${REJSON_BRANCH:-master}
+JSON_REPO_URL="https://github.com/RedisJSON/RedisJSON.git"
+TEST_DEPS_DIR="${ROOT}/tests/deps"
+JSON_MODULE_DIR="${TEST_DEPS_DIR}/RedisJSON"
+JSON_BIN_DIR="${BINROOT}/RedisJSON/${JSON_BRANCH}"
+export JSON_BIN_PATH="${JSON_BIN_DIR}/rejson.so"
+
+# Check if REJSON_PATH is set externally
+if [ -n "$REJSON_PATH" ]; then
+    JSON_BIN_PATH="$REJSON_PATH"
+    echo "Using RedisJSON path given as REJSON_PATH: $REJSON_PATH"
+    return 0
+fi
+
+# Clone the RedisJSON repository if it doesn't exist
+if [ ! -d "${JSON_MODULE_DIR}" ]; then
+    echo "Cloning RedisJSON repository from ${JSON_REPO_URL} to ${JSON_MODULE_DIR}..."
+    run_command git clone --quiet --recursive $JSON_REPO_URL $JSON_MODULE_DIR
+    echo "Done"
+else
+    echo "RedisJSON already exists in ${JSON_MODULE_DIR}"
```

Comment:
Shouldn't we exit if the module was found?

Reply:
See https://github.com/RediSearch/RediSearch/pull/5622#discussion_r1959663019